### ISSUE TITLE
如何通过直接读图来进行训练？

### ISSUE THREAD

**Issue by Ekinkit** [ShawDa/unet-rgb#6]
我按照博主给的代码训练模型，发生了MemoryError，可能是我的数据集比较大加上电脑内存不够，所以我现在想通过直接读取图片的方法来训练模型。请问博主能给一些处理数据的建议吗？

**Comment by jizhang02** [comment]
请问博主的汽车数据集有没有官方来源呢？谢谢

**Comment by ShawDa** [comment]
> 我按照博主给的代码训练模型，发生了MemoryError，可能是我的数据集比较大加上电脑内存不够，所以我现在想通过直接读取图片的方法来训练模型。请问博主能给一些处理数据的建议吗？

你可以选取部分数据集来制作训练集和测试集，并且注意把batch size设最小，这样可以解决内存不够的问题。

**Comment by ShawDa** [comment]
> 请问博主的汽车数据集有没有官方来源呢？谢谢

Carvana Image Masking Challenge：https://www.kaggle.com/c/carvana-image-masking-challenge

**Comment by Ekinkit** [comment]
> > 我按照博主给的代码训练模型，发生了MemoryError，可能是我的数据集比较大加上电脑内存不够，所以我现在想通过直接读取图片的方法来训练模型。请问博主能给一些处理数据的建议吗？
> 
> 你可以选取部分数据集来制作训练集和测试集，并且注意把batch size设最小，这样可以解决内存不够的问题。

谢谢博主 我已经解决了这个问题 还想请问一下：您知道如何计算meanIoU吗？我自己生成的预测图大小和mask图的大小不一致的话，会影响计算吗？

**Comment by ShawDa** [comment]
> > > 我按照博主给的代码训练模型，发生了MemoryError，可能是我的数据集比较大加上电脑内存不够，所以我现在想通过直接读取图片的方法来训练模型。请问博主能给一些处理数据的建议吗？
> > 
> > 
> > 你可以选取部分数据集来制作训练集和测试集，并且注意把batch size设最小，这样可以解决内存不够的问题。
> 
> 谢谢博主 我已经解决了这个问题 还想请问一下：您知道如何计算meanIoU吗？我自己生成的预测图大小和mask图的大小不一致的话，会影响计算吗？

检测IOU计算可以参考这里：https://blog.csdn.net/eddy_zheng/article/details/52126641  预测图大小和mask图大小应该是一样的呀，怎么会不一样了？

**Comment by Ekinkit** [comment]
> > > > 我按照博主给的代码训练模型，发生了MemoryError，可能是我的数据集比较大加上电脑内存不够，所以我现在想通过直接读取图片的方法来训练模型。请问博主能给一些处理数据的建议吗？
> > > 
> > > 
> > > 你可以选取部分数据集来制作训练集和测试集，并且注意把batch size设最小，这样可以解决内存不够的问题。
> > 
> > 
> > 谢谢博主 我已经解决了这个问题 还想请问一下：您知道如何计算meanIoU吗？我自己生成的预测图大小和mask图的大小不一致的话，会影响计算吗？
> 
> 检测IOU计算可以参考这里：https://blog.csdn.net/eddy_zheng/article/details/52126641 预测图大小和mask图大小应该是一样的呀，怎么会不一样了？
我知道啦，因为我生成的预测图已经输出了 并且都固定了尺寸，但其实在计算IoU的时候应该是在模型训练完成、预测的时候（应该是这么理解吧）。
但其实我最想知道的是 我已经了解了meanIoU的计算方法，但是我不知道应该把这个代码加在哪里。
是应该在model.compile 还是fit_generator还是predict？求指点。

**Comment by ShawDa** [comment]
> > > > > 我按照博主给的代码训练模型，发生了MemoryError，可能是我的数据集比较大加上电脑内存不够，所以我现在想通过直接读取图片的方法来训练模型。请问博主能给一些处理数据的建议吗？
> > > > 
> > > > 
> > > > 你可以选取部分数据集来制作训练集和测试集，并且注意把batch size设最小，这样可以解决内存不够的问题。
> > > 
> > > 
> > > 谢谢博主 我已经解决了这个问题 还想请问一下：您知道如何计算meanIoU吗？我自己生成的预测图大小和mask图的大小不一致的话，会影响计算吗？
> > 
> > 
> > 检测IOU计算可以参考这里：https://blog.csdn.net/eddy_zheng/article/details/52126641 预测图大小和mask图大小应该是一样的呀，怎么会不一样了？
> > 我知道啦，因为我生成的预测图已经输出了 并且都固定了尺寸，但其实在计算IoU的时候应该是在模型训练完成、预测的时候（应该是这么理解吧）。
> > 但其实我最想知道的是 我已经了解了meanIoU的计算方法，但是我不知道应该把这个代码加在哪里。
> > 是应该在model.compile 还是fit_generator还是predict？求指点。

可以参考 https://keras.io/zh/models/model/#compile
predict之后你可以和真实情况对比，一般是用在训练集和验证集上，compile,evaluate
